great_expectations/experimental/datasources/data_asset/data_connector/data_connector.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import annotations
+
+import logging
+import pathlib
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+from great_expectations.core.id_dict import BatchSpec
+
+if TYPE_CHECKING:
+    from great_expectations.core.batch import BatchDefinition
+    from great_expectations.experimental.datasources.interfaces import BatchRequest
+
+
+logger = logging.getLogger(__name__)
+
+
+# noinspection SpellCheckingInspection
+class DataConnector:
+    """The base class for all Data Connectors.
+
+    Data Connectors produce identifying information, called Batch Specs, that Execution Engines
+    can use to get individual batches of data. They add flexibility in how to obtain data
+    such as with time-based partitioning, downsampling, or other techniques appropriate
+    for the Datasource.
+
+    For example, a DataConnector could produce a SQL query that logically represents "rows in
+    the Events table with a timestamp on February 7, 2012," which an SqlAlchemy Datasource
+    could use to materialize a SqlAlchemy Dataset corresponding to that Batch of data and
+    ready for validation.
+
+    A Batch is a sample from a data asset, sliced according to a particular rule. For example,
+    an hourly slide of the Events table or “most recent Users records.” It is the primary
+    unit of validation in the Great Expectations Data Context. Batches include metadata that
+    identifies how they were constructed--the same Batch Spec assembled by the data connector.
+    While not every Datasource will enable re-fetching a specific batch of data, GX can store
+    snapshots of batches or store metadata from an external data version control system.
+
+    Args:
+        datasource_name: The name of the Datasource associated with this DataConnector instance
+        data_asset_name: The name of the DataAsset using this DataConnector instance
+    """
+
+    def __init__(
+        self,
+        datasource_name: str,
+        data_asset_name: str,
+    ) -> None:
+        self._datasource_name: str = datasource_name
+        self._data_asset_name: str = data_asset_name
+
+        # This is a dictionary which maps data_references onto batch_requests.
+        self._data_references_cache: Dict[str, List[BatchDefinition] | None] = {}
+
+        self._data_context_root_directory: Optional[pathlib.Path] = None
+
+    @property
+    def data_asset_name(self) -> str:
```

Comment:
Is it one data connector per data asset, or one data connector per datasource? If the latter, this property should be removed

Reply:
Since datasources are connector specific (ex: Pandas S3), I'd expect data_asset_name should not be here

Reply:
@donaldheppner We have one `DataConnector` per `DataAsset`; since `DataAsset` utilizes `data_asset_name` today, it also needs to be passed to the new `DataConnector` to handle the same functionality in that, higher, level.  Happy to discuss.  Thanks.